packages/query/src/block-processor.ts

```diff
@@ -276,15 +295,22 @@ export class BlockProcessor implements BlockProcessorInterface {
       const assetId = n.note?.value?.assetId;
       if (!assetId) continue;
 
-      const metadataInDb = await this.indexedDb.getAssetsMetadata(assetId);
-      if (metadataInDb) continue;
+      await this.saveAndReturnMetadata(assetId);
+    }
+  }
+
+  private async saveAndReturnMetadata(assetId: AssetId): Promise<Metadata | undefined> {
+    const metadataAlreadyInDb = await this.indexedDb.getAssetsMetadata(assetId);
+    if (metadataAlreadyInDb) return metadataAlreadyInDb;
 
-      const metadataFromNode = await this.querier.shieldedPool.assetMetadata(assetId);
+    const metadataFromNode = await this.querier.shieldedPool.assetMetadataById(assetId);
 
-      if (metadataFromNode) {
-        await this.indexedDb.saveAssetsMetadata(customizeSymbol(metadataFromNode));
-      }
+    if (metadataFromNode) {
+      await this.indexedDb.saveAssetsMetadata(customizeSymbol(metadataFromNode));
+      return metadataFromNode;
     }
+
+    return undefined;
```

Comment:
Just extracting this logic out into a new `saveAndReturnMetadata` method so that it can be reused below.